src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixTask.groovy

```diff
@@ -65,36 +71,47 @@ class ScalafixTask extends SourceTask {
                   | - Classpath: ${classpath}
                   |""".stripMargin())
 
-        def classloader = this.class.classLoader
-        def customRulesClassloader = classloaderFrom(customRulesConfiguration, classloader)
-
-        def args = Scalafix.classloadInstance(classloader)
+        def classloader = Scalafix.class.classLoader
+        def externalRulesClassloader = classloaderFrom(customRulesConfiguration, classloader)
+        def scalafixArgs = Scalafix.classloadInstance(classloader)
                 .newArguments()
                 .withMode(mode)
                 .withConfig(configFilePath)
                 .withRules(rules.get())
                 .withSourceroot(Paths.get(sourceRoot))
                 .withPaths(sourcePaths)
-                .withToolClasspath(customRulesClassloader)
+                .withToolClasspath(externalRulesClassloader)
                 .withClasspath((classpath ?: []).collect { Paths.get(it)} )
                 .withScalaVersion(scalaVersion)
                 .withScalacOptions(compileOptions)
 
         logger.debug(
                 """Scalafix rules:
-                  | - Available: ${args.availableRules().collect { it.name() }}
-                  | - That will run: ${args.rulesThatWillRun().collect { it.name() }}
+                  | - Available: ${scalafixArgs.availableRules().collect { it.name() }}
+                  | - That will run: ${scalafixArgs.rulesThatWillRun().collect { it.name() }}
                   |""".stripMargin())
 
-        if (!args.rulesThatWillRun().empty) {
+        if (!scalafixArgs.rulesThatWillRun().empty) {
+            assertSemanticdbIsConfigured(scalafixArgs.rulesThatWillRun())
+
             logger.quiet("Running Scalafix on ${sourcePaths.size} Scala source file(s)")
-            def errors = args.run()
+            def errors = scalafixArgs.run()
             if (errors.size() > 0) throw new ScalafixFailed(errors)
         } else {
             logger.warn("No Scalafix rules to run")
         }
     }
 
+    private void assertSemanticdbIsConfigured(List<ScalafixRule> rulesThatWillRun) {
+        def semanticRules = rulesThatWillRun.findAll { it.kind().isSemantic() }
+
+        if (!semanticRules.empty && !semanticdbConfigured) {
+            def semanticRuleNames = semanticRules.collect { it.name() }.join(", ")
+            throw new GradleException("The semanticdb compiler plugin is required to run semantic rules such as ${semanticRuleNames}. " +
+                    "To fix this problem, please enable 'autoConfigureSemanticdb' in the scalafix plugin extension")
+        }
+    }
```

Comment:
as a side effect of including the dependency jars in the classpath passed to Scalafix is that the latter no longer reports a `ScalafixError.MissingSemanticdbError` when running semantic rules without having the Semanticdb files generated by the compiler plugin. The `sbt-scalafix` plugin performs some [upfront validation](https://github.com/scalacenter/sbt-scalafix/blob/master/src/main/scala/scalafix/internal/sbt/SemanticRuleValidator.scala#L18-L19) like I'm doing now ☝️ . I managed to bypass the Sbt plugin validation by adding `semanticdb-scalac` as a dependency of the project and got the same result: no `ScalafixError.MissingSemanticdbError` is reported.